terraform/ec2/main.tf

```diff
@@ -41,7 +41,7 @@ module "basic_components" {
 
   cortex_instance_endpoint = var.cortex_instance_endpoint
 
-  sample_app_listen_address_host = aws_instance.sidecar.public_ip
+  sample_app_listen_address_host = one(aws_instance.sidecar[*].public_ip)
```

Comment:
Is the usage of `one()` going to break our `Canary` tests since we are [pinned](https://github.com/aws-observability/aws-otel-collector/blob/e4ddd80fffefca1bb417ed07996a508931ec96de/.github/workflows/canary.yml#L112) to version 14? `one` appears to require `v15` or newer. [docs](https://developer.hashicorp.com/terraform/language/functions/one).